### PR TITLE
Add `chainerx.argmin` and `chainerx.ndarray.argmin`

### PR DESCRIPTION
**Pull request by Harshan01**
With respect to #6623, ArgMin in ChainerX is implemented.

Due to recent changes #6675 , the ArgMin ChainerX implementation is hard-coded for simplicity and to avoid further complications.

Native and CUDA Device implementation files are updated, namely native_device/reduction.cc and cuda_device/reduction.cu

### CODE REVIEW

**Comment by Harshan01** [comment]
Ref issue: #6423

**Comment by niboshi** [comment]
Thank you for the PR! Implementation looks good.
Jenkins, test this please.

**Comment by niboshi** [comment]
It seems `TestArgmin` have failed. Could you check the results? @Harshan01

**Comment by Harshan01** [comment]
I am checking it out

**Comment by Harshan01** [comment]
@niboshi , I have removed the conflicts but the build for MacOS fails for some unknown reason. Can you help me out?

**Comment by niboshi** [comment]
@Harshan01 
Thanks!
The test failure is just an occasional one we're working on. Please ignore it.

Jenkins, test this please

**Comment by Harshan01** [comment]
@niboshi , test in py27 and py35 does not seem related to ArgMin, but in py37, the testcases fail, though the same testcases pass in other checks. Could you give some insights in this issue?

File: chainerx_cc/chainerx/cuda/cuda_device/reduction.cu [hunk 1]
```diff
@@ -60,6 +60,46 @@ public:
...
+struct ArgMinImpl {
+    using CudaType = cuda_internal::DataType<T>;
+    struct MinAndArgMin {
+        CudaType min;
+        int64_t argmin;
+    };
+    __device__ MinAndArgMin Identity() { return {CudaType{}, -1}; }
+    __device__ MinAndArgMin MapIn(CudaType in, int64_t index) { return {in, index}; }
+    __device__ void Reduce(MinAndArgMin next, MinAndArgMin& accum) {
+        if (accum.argmin < 0 || accum.min > next.min) {
```
Comment (niboshi) [inline chainerx_cc/chainerx/cuda/cuda_device/reduction.cu]:
```suggestion
        if (next.argmin != -1 && (accum.argmin == -1 || accum.min > next.min)) {
```
How about this fix to match `ArgMax` implementation?

**Comment by Harshan01** [comment]
Applied suggested changes.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by Harshan01** [comment]
@niboshi , I have resolved the conflicts. Can you check it out?

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM
Thank you so much!

**Pull request merged**